### PR TITLE
Align Adjust to spec

### PR DESCRIPTION
**Pull request by patrick-tolosa**
#### What does this PR solve?
Align Adjust to Spec

- Remove tint (Not in Spec)
- Add ImproveMode Param (Enum of two values)
- Add recolor()
- Add replaceColor
- Add fillLight()
- Fix improve()
- Add a fix to ParamValue to detect falsish values  correctly (such as 0)

#### Final checklist
- [X] JSdoc - Add proper docstrings based on our PHP implementation
- [X] JSdoc - Hide private functions using @private
- [X] Implementation is aligned with Spec
- [X] Tests - Add proper tests to the added code

### CODE REVIEW

File: src/actions/adjust/Adjust.ts [hunk 1]
```diff
@@ -215,10 +208,37 @@ function vibrance(strength?:number): EffectAction {
...
+function replaceColor(toColor:string): ReplaceColorAction {
+  return new ReplaceColorAction(toColor);
+}
+
+/**
+ * @description Converts the colors of every pixel in an image based on the supplied color matrix, in which the value of each color channel is calculated based on the values from all other channels (e.g. a 3x3 matrix for RGB, a 4x4 matrix for RGBA or CMYK, etc).</br>
+ * For every pixel in the image, take each color channel and adjust its value by the specified values of the matrix to get a new value.
+ * @param {number[][]} matrix
+ */
+function recolor(matrix:number[][]): RecolorAction {
```
Comment (strausr) [inline src/actions/adjust/Adjust.ts:226]: I know this is not in the spec but does it not look weird that recolor is not in camalcase?

File: src/params/improveMode/ImproveMode.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const OUTDOOR = 'outdoor';
```
Comment (strausr) [inline src/params/improveMode/ImproveMode.ts:1]: Should this be under constants rather than params?

**Review by strausr — CHANGES_REQUESTED** [review]

File: src/params/improveMode/ImproveMode.ts (hunk 2 above)
Comment (patrick-tolosa) [reply src/params/improveMode/ImproveMode.ts:1]: I'm not sure it makes sense to keep both Constants and Params, one of them has to go (And we also need a better name).

I think everything that comes from the "outside" needs to go Params

File: src/actions/adjust/Adjust.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrick-tolosa) [reply src/actions/adjust/Adjust.ts:226]: As per spec, let's open a bug(or change request) to the spec if you feel it's needed.

File: __TESTS_BUNDLE_SIZE__/bundleSizeTestCases.ts [hunk 3]
```diff
@@ -22,8 +22,8 @@ const bundleSizeTestCases:ITestCase[] = [
     ]
   },
   {
-    name: 'Tests a transformable image with Resize and just',
-    sizeLimitInKB: 11,
+    name: 'Tests a transformable image with Resize and ajust',
```
Comment (maoznir) [inline __TESTS_BUNDLE_SIZE__/bundleSizeTestCases.ts:25]: change to "Adjust"?

Comment (patrick-tolosa) [reply __TESTS_BUNDLE_SIZE__/bundleSizeTestCases.ts]: Done, also capitzlized Border and Quality.

**Review by strausr — APPROVED** [review]

**Review by maoznir — APPROVED** [review]

**Pull request merged**